File: 3206_PhotoLesson_iOS/Phase 1/3206_PhotoLesson/Models/Payment.swift
import Foundation

struct PaymentRequest: Codable {
    let courseId: Int
}

struct PaymentRequestResponse: Codable {
    let orderId: String
    let amount: Int
    let clientKey: String
    let orderName: String
    let courseId: Int
}

struct PaymentConfirm: Codable {
    let paymentKey: String
    let orderId: String
    let amount: Int
}

struct PaymentResponse: Codable {
    let paymentId: Int
    let memberId: Int
    let memberName: String?
    let courseId: Int
    let courseTitle: String?
    let orderId: String
    let amount: Int
    let status: String
    let method: String?
    let receiptUrl: String?
    let createdAt: String?
}

File: 3206_PhotoLesson_Backend/Phase 1/src/main/java/com/photolesson/backend/entity/Payment.java
package com.photolesson.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "payments", indexes = {
        @Index(name = "idx_payment_member", columnList = "member_id"),
        @Index(name = "idx_payment_course", columnList = "course_id"),
        @Index(name = "idx_payment_status", columnList = "status")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id", nullable = false)
    private Course course;

    @Column(name = "order_id", unique = true, nullable = false, length = 100)
    private String orderId;

    @Column(name = "payment_key", unique = true, length = 200)
    private String paymentKey;

    @Column(nullable = false)
    private Integer amount;

    @Column(length = 20, nullable = false)
    @Builder.Default
    private String status = "PENDING"; // PENDING, SUCCESS, FAILED, REFUNDED

    @Column(length = 50)
    private String method; // 카드, 계좌이체 등

    @Column(name = "receipt_url", length = 500)
    private String receiptUrl;

    @Column(name = "fail_reason", length = 500)
    private String failReason;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
